source/backend/api/Startup.cs

```diff
@@ -342,6 +343,10 @@ public void Configure(IApplicationBuilder app, IWebHostEnvironment env, IApiVers
 
             app.UseRouting();
             app.UseCors();
+
+            // Set responses secure headers.
+            ConfigureSecureHeaders(app, Configuration);
```

Comment:
So, from what I can see this will result in two CSPs, one from the api and one from nginx. The one on the index has been removed.

I'm still having a hard time conceptualizing how the two CSPs will interact. presumably the nginx one will take effect first since the user will hit the nginx box before they are even served the react app and therefore also before they get this csp from the api. If that is the case, what is the point of this CSP? does this CSP override the nginx one because it happens later in the startup process? What is the benefit to having two CSPs? can we get away with just having one?

sorry, a bunch of questions I know but still confused about this situation.

Reply:
I think it depends on what is serving the response[1]. If the response is being served by Nginx then the CSP policy on the headers will be what was set for the front end. If the reply is coming from the backend, then the backend will set the response headers. I am not 100% sure about the mechanisms we are using for proxying the backend calls, but it does not look like the headers are being overridden.

Also, I don't think the CSP is necessary for APIs given that there is no rendering on the front end. But having it set explicitly makes it more clear.

In [1] you can see the last request with the changes from this PR and the explicit CSP from the backend api.
[1] 
![image](https://user-images.githubusercontent.com/12201292/220800008-d97e389c-8655-48bd-b209-8d741ba4ad57.png)